src/lib/routes/admin-api/event.ts

```diff
@@ -2,26 +2,95 @@ import { Request, Response } from 'express';
 import { IUnleashServices } from '../../types/services';
 import EventService from '../../services/event-service';
-import { ADMIN } from '../../types/permissions';
+import { ADMIN, NONE } from '../../types/permissions';
 import { IEvent } from '../../types/events';
 import Controller from '../controller';
 import { anonymise } from '../../util/anonymise';
+import { OpenApiService } from '../../services/openapi-service';
+import { createResponseSchema } from '../../../lib/openapi';
+import {
+    eventsSchema,
+    EventsSchema,
+} from '../../../lib/openapi/spec/events-schema';
+import { serializeDates } from '../../../lib/types/serialize-dates';
+import {
+    featureEventsSchema,
+    FeatureEventsSchema,
+} from '../../../lib/openapi/spec/feature-events-schema';
+import { getStandardResponses } from '../../../lib/openapi/standard-responses';
 
 const version = 1;
 export default class EventController extends Controller {
     private eventService: EventService;
 
     private anonymise: boolean = false;
 
+    private openApiService: OpenApiService;
+
     constructor(
         config: IUnleashConfig,
-        { eventService }: Pick<IUnleashServices, 'eventService'>,
+        {
+            eventService,
+            openApiService,
+        }: Pick<IUnleashServices, 'eventService' | 'openApiService'>,
     ) {
         super(config);
         this.eventService = eventService;
         this.anonymise = config.experimental?.anonymiseEventLog;
-        this.get('/', this.getEvents, ADMIN);
-        this.get('/:name', this.getEventsForToggle);
+        this.openApiService = openApiService;
+
+        this.route({
+            method: 'get',
+            path: '',
+            handler: this.getEvents,
+            permission: ADMIN,
+            middleware: [
+                openApiService.validPath({
+                    operationId: 'getEvents',
+                    tags: ['admin'],
+                    responses: {
+                        ...getStandardResponses(401),
+                        200: createResponseSchema('eventsSchema'),
+                    },
+
+                    parameters: [
+                        {
+                            name: 'project',
+                            description:
+                                'The name of the project whose events you want to retrieve',
+                            schema: { type: 'string' },
+                            in: 'query',
+                        },
+                    ],
+
+                    description:
+                        'Returns **the last 100** from the Unleash instance when called without a query parameter. When called with a `project` parameter, returns **all events** for the specified project.\n\nIf the provided project does not exist, the list of events will be empty.',
+                    summary:
+                        'Get the most recent events from the Unleash instance or all events related to a project.',
```

Comment:
Nice work! I think if we should move the long texts somewhere else though, to not clutter the code.

Maybe we could move the text to a seprate TS or JSON file? Or maybe top-level constants.

Reply:
Yeah, a separate "endpoint-descriptions.ts" file might be a good idea.
I'll see what I can do.